File: csharp/decorator_practice/decorator_practice/test/ApproveTest.cs
using System;
using System.Collections.Generic;
using System.Text;
using decorator_practice.main.decorator;
using decorator_practice.main.privileges;
using decorator_practice.main.users;
using Moq;
using NUnit.Framework;

namespace decorator_practice.test
{
    [TestFixture]
    class ApproveTest
    {
        public Mock<User> user;

        [SetUp]
        public void SetUp()
        {
            user = new Mock<User>();
        }

        [TestCase]
        public void NameTest()
        {
            User approve = new Approve(user.Object);
            Assert.AreEqual("Any Name approve", user.Object.getPrivileges());
        }
    }
}

File: java/src/test/java/mx/iteso/privileges/ApproveTest.java
package mx.iteso.privileges;

import mx.iteso.User;
import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.assertEquals;


public class ApproveTest  {
    private User user;
    @Before
    public void setUp() {
        user = new FinalUser();
    }

    @Test
    public void testApprove() {
        assertEquals(user.approve(), "No privileges to approve");
        this.user = new Approve(this.user);
        assertEquals(user.approve(), "Can approve");
    }

}
